app/CoinDataStorage.hs

```diff
@@ -24,28 +26,53 @@ initializeDB dbLoc = DB.open (toText dbLoc) >>= \db ->
   return db
 
 insertCoins :: DB.Database -> Vector Coin -> IO ()
-insertCoins db vc = DB.exec db $ _buildInsertStatemnt vc
+insertCoins db vc = catch(_exInsertCoins db vc) handleIt where
+  handleIt e =
+    case fromException e of
+      Just (DB.SQLError code details context) -> 
+        case (code,_matchErr _missingColumnErr details) of 
+          (DB.ErrorError,idx:_) -> -- attempt to handle schema changes
+            let sndHandler (SomeException e) = throwIO (CoinDataStorageError _insertFailureMsg) in
+            catch (__clearAndInit db) sndHandler >> -- attempt to drop tables and reinit db
```

Comment:
For the longer turn, I think we may not want to clear the DB during a refresh implicitly. It will be nice to handle the error on the Views, then prompt the user if they will like to clear/reset the DB.

Reply:
Yep that sounds like the best approach.